src/Fuzz.elm

```diff
@@ -357,13 +365,33 @@ listShrinkHelp listOfTrees =
         root =
             List.map RoseTree.root listOfTrees
 
+        dropFirstHalf : List (RoseTree a) -> RoseTree (List a)
+        dropFirstHalf list_ =
+            List.drop (List.length list_ // 2) list_
+                |> listShrinkRecurse
+
+        dropSecondHalf : List (RoseTree a) -> RoseTree (List a)
+        dropSecondHalf list_ =
+            List.take (List.length list_ // 2) list_
+                |> listShrinkRecurse
+
+        halved : LazyList (RoseTree (List a))
+        halved =
+            if n >= 4 then
+                Lazy.lazy <|
+                    \_ ->
+                        Lazy.List.fromList [ dropFirstHalf listOfTrees, dropSecondHalf listOfTrees ]
+                            |> Lazy.force
+            else
+                Lazy.List.empty
```

Comment:
This logic comes in addition to and not instead of the existing 'remove one element' and 'shrink one element' strategies. It adds an extra high-reward strategy as the first thing to try, but always falling back to the more fine-grained strategies if that doesn't work. So the minimal counterexample you mention should still be found, and I checked it is.

This conditional merely says shrinking by halving should not be attempted for lists smaller than 4 elements, shrinking by a single element is still attempted like before. The reason I added this conditional: Once we're at lists sizes smaller than 4 'halving' might sometimes simply remove a single element, which is something the algorithm was already going to try in the 'remove one' phase anyway. To prevent us from trying the same shrink twice this conditional was introduced.